### ISSUE TITLE
docker containers with development environments

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#504]
I've been playing around with docker and exploring how best to use it to set up easy to use environments.

I already used it to create a containerized multi-tier architecture with separate server and analysis tiers (https://github.com/e-mission/e-mission-docker/pull/4). @xubowenhaoren has tried it out and said that it works well.

I am now looking into dockerizing the development environments (e.g. https://github.com/e-mission/e-mission-docker/pull/5). I'm working on the devapp first. Is there anybody who wants to try it out and give me feedback on the requirements?

**Comment by shankari** [comment]
In particular, I think that people want to be able to edit the files from a host directory rather than within docker. We can use volumes to map a host directory to a container directory. But should we assume that the user wants to checkout the code directory, or should the container do it for the user? Presumably the user wants to checkout code, switch between branches, etc while developing.

**Comment by xubowenhaoren** [comment]
Yes, I would love to have the option to configure the branch and other config files outside of the docker.

**Comment by shankari** [comment]
@xubowenhaoren just to clarify, the goal is to make it easier to use docker to set up *development* environments, not production environments.

**Comment by shankari** [comment]
I now have a docker image for the devapp-server. It works in two modes:
- `run`: for launching from any branch to try out the UI
- `develop`: mounts a host directory to edit and develop UI code

@xubowenhaoren @stephhuerre @PatGendre @kafitz @atton16
can you please try it out and see if it works for you and whether the README makes sense?
https://github.com/e-mission/e-mission-docker/pull/5

I am going to merge anyway in the next couple of days.

**Comment by PatGendre** [comment]
Hi @shankari I tested on my PC (not on PWD) the notebook version, and it worked fine :-)
The readme is easy to use, just one detail, in the main [README](https://github.com/e-mission/e-mission-docker/blob/master/README.md) there is a sample instruction on how to load data on the server, this instruction would be helpful in the [notebook README](https://github.com/e-mission/e-mission-docker/blob/master/examples/juypter-notebook/README.md) as well.
It is a good idea to update the docker image, thanks!

**Comment by shankari** [comment]
@PatGendre the notebook container has been around forever. Did you get a chance to test the pending change, which creates containers for the UI?

I was thinking of the notebook README as a supplement to the main README so that we wouldn't copy the information everywhere. If we needed to change it, then if it was in two places, it would need to be changed in two places.

Also, wrt loading data onto the server, wait for my next set of updates :)

**Comment by PatGendre** [comment]
@shankari Ok thanks, I didn't get your point, now it is clearer.    
As for the devapp-serve container, it is definitely interesting, I can install it, but for the moment I've never worked on the UI and didn't even install android studio :-(     
I should test this, we should have an opportunity for a new e-mission deployment in the coming weeks, so I would see that with Yann @iciatech-yprime. I'll continue to have a look of the next updates;-)
